File: typescript/src/nodes/catalogue/logNode.ts
import { IAgentContext } from "@/agent";
import { BaseNode, NodeResult } from "@/nodes";

class LogNode extends BaseNode {
  id = "logNode";

  async run(ctx: IAgentContext): Promise<NodeResult | void> {
    const previousResult = ctx.nodeResults?.at(-1);
    return {
      nodeId: this.id,
      value: `logged: ${JSON.stringify(previousResult?.value)}`,
    };
  }
}

export default new LogNode();

File: python/nodes/catalogue/log_node.py
from __future__ import annotations

from typing import Optional

from agentic.agent.core.context import AgentContext
from agentic.nodes.core.node import BaseNode, NodeResult


class LogNode(BaseNode):
    id = "logNode"

    async def run(self, ctx: AgentContext) -> Optional[NodeResult]:
        previous_result = ctx.node_results[-1] if ctx.node_results else None
        return NodeResult(
            node_id=self.id,
            value=f"logged: {previous_result.value if previous_result else None}",
        )


log_node = LogNode()
